### DISCUSSION TITLE
Scan Upload/Library Folder For New Files

### DISCUSSION THREAD

**Discussion by xCJPECKOVERx** [immich-app/immich#13885] [Q&A]
I'm looking at transferring my Photoprism library to Immich. I've played around with setting up an external library to my current Photoprism Originals, but I'd like to transfer everything to be directly controlled by Immich, in one location, so the whole library is contained within upload/library/user.

Is there any way to make Immich scan it's library or uploads folder for new photos and index them? Similar to how Photoprism can be forced to index its Originals folder? Having to upload many thousands of photos through the website/mobile app would be pretty cumbersome, especially when I can just move them in the file system directly.

**Comment by schuhbacca** [comment]
You cannot scan the upload location, it is meant to be completely controlled by immich. 

You are probably best off using the CLI to upload everything or use an external library.

**Reply by atb00ker** [reply]
~If I "upload" using CLI, would it duplicate the files or change the permissions of the existing files?
Context: I've already pasted a large data dump in the immich's "/upload" location (in the correct locations with existing images) and now I am trying to make immich seen these new images as if I uploaded them using the Web UI.~

Nevermind, I poked around the code a wee bit and it looks like cli upload tool is just throwing requests to the api and actually reuploading instead of just adding them to the database.

I think I'll just upload the images to a dummy location, when upload is done, I'll close the server and switch the location my current folder to fool immich to think that it uploaded in the existing location and call it a day. (My use case is onetime upload anyway)

**Reply by gotjoshua** [reply]
did this switcheroo technique work out for you @atb00ker ?

**Reply by atb00ker** [reply]
It's been a while, I don't remember. 
I think I ended up uploading all the images from the API.

https://api.immich.app/endpoints/assets/uploadAsset

**Reply by gotjoshua** [reply]
me too 😿
